File: Spring/Task 2 (Hierarchy)/sources/ice_cream/fruit_ice/fruit_ice.cpp
//
// Created by rinsler on 26.03.18.
//

#include "fruit_ice.h"


FruitIce::FruitIce(std::string name, std::string manufacturer, Ingredient juice) :
        IceCream(name, manufacturer),
        juice(juice)
{

}


std::string FruitIce::getRecipe()
{
    return "Recipe of " + name +
           " from " + manufacturer +
           ": freeze " + std::to_string(juice.weight) +
           " grams of " + juice.name +
           " juice.";
}

File: Spring/Task 2 (IceCream)/IceCreamSources/IceCreamImplementations/FruitIce.java
package IceCreamImplementations;

import IceCreamAbstract.IceCream;

public class FruitIce extends IceCream {

    protected Ingredient juice;

    public FruitIce(String name, String manufacturer, Ingredient juice) {
        super(name, manufacturer);
        this.juice = juice;
    }

    @Override
    public String getRecipe() {
        return "Recipe of " + name +
                " from " + manufacturer +
                ": freeze " + juice.getWeight() +
                " grams of " + juice.getName() +
                " juice.";
    }

    public Ingredient getJuice() {
        return juice;
    }

    public void setJuice(Ingredient juice) {
        this.juice = juice;
    }
}
